test/regression/aarch64/Syscall.cc

```diff
@@ -872,6 +872,50 @@ TEST_P(Syscall, ftruncate) {
   EXPECT_EQ(getGeneralRegister<uint64_t>(23), 0);
 }
 
+TEST_P(Syscall, sysinfo) {
+  RUN_AARCH64(R"(
+    # Get heap address
+    mov x0, 0
+    mov x8, 214
+    svc #0
+    mov x20, x0
+
+    mov x1, #10000
+    mov x2, #10
+    mul x1, x1, x2
+
+    .loop:
+    subs x1, x1, #1
+    b.ne .loop
+
+    # syscall(addr)
+    mov x0, x20
+    mov x8, #179
+    svc #0
+    mov x21, x0
+  )");
+
+  // Check returned 0
+  EXPECT_EQ(getGeneralRegister<uint64_t>(21), 0);
+
+  // Check outputted sysinfo struct
+  uint64_t paddr = process_->translate(process_->getHeapStart());
+  // The size of the sysinfo struct within SimEng is 102 bytes
+  char reference[102] = {0};
```

Comment:
Can use `std::array<char, 102>`

Reply:
you might also have to add `std::fill(reference.begin(), reference.end() 0);`

Reply:
Any advantage to using std::array over char[] ?

Reply:
I mean aren't we actively trying to not use C-style declarations?

Reply:
yeh good point, changing